plans/action/install.pp

```diff
@@ -203,27 +208,32 @@
   # Create csr_attributes.yaml files for the nodes that need them. Ensure that
   # if a csr_attributes.yaml file is already present, the values we need are
   # merged with the existing values.
-
-  run_plan('peadm::util::insert_csr_extension_requests', $master_target,
-    extension_requests => {
-      peadm::oid('peadm_role')               => 'puppet/master',
-      peadm::oid('peadm_availability_group') => 'A',
-    },
-  )
-
-  run_plan('peadm::util::insert_csr_extension_requests', $puppetdb_database_target,
-    extension_requests => {
-      peadm::oid('peadm_role')               => 'puppet/puppetdb-database',
-      peadm::oid('peadm_availability_group') => 'A',
-    },
-  )
-
-  run_plan('peadm::util::insert_csr_extension_requests', $puppetdb_database_replica_target,
-    extension_requests => {
-      peadm::oid('peadm_role')               => 'puppet/puppetdb-database',
-      peadm::oid('peadm_availability_group') => 'B',
-    },
-  )
+  parallelize([$master_target, $puppetdb_database_target, $puppetdb_database_replica_target]) |$target| {
+    if ($target[0].name == $master_target[0].name) {
```

Comment:
This is probably unsafe for non-XL scenarios in which `$puppetdb_database_target` and/or `$puppetdb_database_replica_target` are empty arrays ( `[ ]` ).

Does Puppet allow us to just make array comparisons directly, and so avoid indexing in or calling methods on objects that may not be there?

If we want to address empty arrays specifically, we can use `.peadm::flatten_compact()` on the argument to `parallelize()`

Reply:
You're probably correct about the empty arrays, I forgot to test other architectures. Will do so using `peadm::flatten_compat()`

As for the indexing, it allowed it but it didn't result in what you'd expect. The result was that the primary database wouldn't get any extensions set and replica would be classified as the primary. It was very strange and given how long it takes to test, I chose to move forward with the direct comparison of the string value. To get to the bottom of what is going on it'll simply take more time as I'll need to modify peadm to short circuit the stuff that takes really long but still does everything to prep for those long bits.

I'll go ahead and double check my work on the equality tests when testing other architectures, just in case I overlooked something; I had actually been testing out a couple patterns so can't say with 100% certainty that I didn't do something on accident that caused the observed behavior.